napari_aicsimageio/core.py

```diff
@@ -7,13 +7,35 @@
 import xarray as xr
 from aicsimageio import AICSImage, exceptions, types
 from aicsimageio.dimensions import DimensionNames
+from qtpy.QtWidgets import QListWidget
+from qtpy.QtCore import Qt
+from napari import Viewer
 
 ###############################################################################
 
 LayerData = Union[Tuple[types.ArrayLike, Dict[str, Any], str]]
 PathLike = Union[str, List[str]]
 ReaderFunction = Callable[[PathLike], List[LayerData]]
 
+###############################################################################
+# _get_viewer() function from https://github.com/napari/napari/issues/2202
+# To provide access to the napari viewer to make the dock widget
+# Copyright (c) 2021 Jonas Windhager
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+def _get_viewer() -> Optional[Viewer]:
+    import inspect
+
+    frame = inspect.currentframe().f_back
+    while frame:
+        instance = frame.f_locals.get("self")
+        if instance is not None and isinstance(instance, Viewer):
+            return instance
+        frame = frame.f_back
+    return None
```

Comment:
This does seem weird but I does also work..... Going to ask some members of the team for thoughts

Reply:
yeah, it's nifty in the coding sense, but probably a bit expensive and error prone.
instead of this `_get_viewer()`, you might be better off with `napari.current_viewer()` (requires 0.4.11).  It doesn't depend on stack inspection, and will _probably_ always be the viewer you want (with perhaps the exception of cases where someone has multiple viewers open, and then _programmatically_ opens a file with the non-current viewer)... but this is all so "experimental" that that's probably the least of our concerns here :)

Reply:
+1 for `napari.current_viewer()`! Thanks @tlambert03

Reply:
Thanks! I realize I was using a napari 4.10 env. I'll update and give that a go.

Reply:
Please update the `setup.py` to the new min version too.

Reply:
I was just about to ask about the versioning! Thanks for the pointer, I think I can figure out.

Reply:
`viewer = napari.current_viewer()` works perfectly.
How are versions handled? Will folks with 4.10 get the old behavior (ie. the current version of this plugin?)? Should I try to wrap this in some `try/except` code with a warning that scene selection requires 4.10?

Reply:
Folks with `napari~=4.10` wont be able to install the new version of the plugin. By updating it in our setup.py you are saying "this is the minimum required napari version" (longform the `~=` operator equates to "anything greater than 0.4.11 and less than 0.5"

no need to add a try except